### PR TITLE
[frontend] buyer, seller, admin, shop image & name

### LINKED ISSUE #112: upload image cropping and accept only image file
**Is your feature request related to a problem? Please describe.**
The upload icon function on ``edit shop info``, ``edit user info`` and possibly ``edit product`` have some problem:
- User can choose any file type to upload ( in the ``edit shop info`` and ``edit user info``, i block it but it's after it uploaded
- The image may be too large, too small or have wrong ratio.

**Describe the solution you'd like**
- Accept only image file:
  - find a why to only allow image type when choose file to upload
- Image cropping
  - use css to handle showing the image on icon, not just ``weight:100%``
  - process the uploaded image, crop it or scale it if too big or small.

### PR DESCRIPTION
**Pull request by KatoToyoki**
Resolveeeeeeeees part of #94

### admin (login with admin)
- admin index page
  image

### buyer (login with customer)
- buyer index page
  image

### seller (login with customer)
- seller index page
  image
- click the `view shop` button will transfer to the shop page

### shop (login with customer)
- after transfer to the shop page
  image
- the two buttons on left side are clickable and will lead to correct path
  image
- you can also try type url by yourself `http://localhost:8080/user2/`, also try the buttons
  image
- note that the `products` inside the shop page is not correct, only the user image, name and url are correct  because at that time i use the wrong api, that is fixed by @UGisBusy, and will not appear in this pr 

### Test Environment Setup
open 3 terminals
one for `docker-compose up postgres minio `
one for `go run .`
one for `cd frontend` -> `npm i` -> `npm run build`
then open the browser then enter http://localhost:8080/ 

### Additional context

N/A

### CODE REVIEW

**Comment by UGisBusy** [comment]
i have fixed the shop part of the icon and shop name on #141

**Comment by KatoToyoki** [comment]
> i have fixed the shop part of the icon and shop name on #141

should i remove it?

File: frontend/src/pages/user/shop/index.tsx
Comment (UGisBusy) [inline frontend/src/pages/user/shop/index.tsx:1]: revert this change because i yoinked it into my pr

**Review by UGisBusy — COMMENTED** [review]
lgtm other than the no lgtm part

File: frontend/src/lib/Auth.ts [hunk 1]
```diff
@@ -18,3 +24,13 @@ export const useAuth = () => {
   const { tokenRef } = useContext(AuthContext);
   return tokenRef.current;
```
Comment (NOOBDY) [inline frontend/src/lib/Auth.ts:25]:
```suggestion
  const token = jwtDecode<CustomJwtPayload>(tokenRef.current);
  return {
    tokenString: tokenRef.current,
    token: token,
  };
```
I think doing this way would be cleaner but idk how much code would this break
you could close it if you don't want to change this

File: frontend/src/lib/Auth.ts [hunk 2]
```diff
@@ -1,5 +1,11 @@
 import { useContext } from 'react';
 import { AuthContext } from '@components/AuthProvider';
+import { jwtDecode, JwtPayload } from 'jwt-decode';
+
+interface CustomJwtPayload extends JwtPayload {
+  role?: string;
+  username?: string;
```
Comment (NOOBDY) [inline frontend/src/lib/Auth.ts]:
```suggestion
  role: string;
  username: string;
```

if the token exists, shouldn't the values always be present?

File: frontend/src/lib/Auth.ts (hunk 1 above)
Comment (KatoToyoki) [reply frontend/src/lib/Auth.ts:25]: if it really needs to modify to this, i think maybe leave for the last because it required all request header that need `useAuth()`  changed

**Review by NOOBDY — APPROVED** [review]

**Review by Error0229 — APPROVED** [review]
Works on my machine, except for the issues that will be fixed in #141.
